### PR TITLE
Clicking new conversation on finalized works

### PR DESCRIPTION
**Pull request by chrisnojima**
This fixes clicking on the new conversation link on finalized conversations. 
To test this you can just have a convo w/ a test account then reset it

Includes some go code from @mmaxim

### CODE REVIEW

File: shared/actions/chat.js [hunk 1]
```diff
@@ -205,6 +206,10 @@ function _pendingToRealConversation (oldKey: ConversationIDKey, newKey: Conversa
   return {payload: {newKey, oldKey}, type: 'chat:pendingToRealConversation'}
 }
 
+function _replaceConversation (oldKey: ConversationIDKey, newKey: ConversationIDKey): ReplaceConversation {
```
Comment (chrisnojima) [inline shared/actions/chat.js]: we can replace conversations that aren't pending -> real. now it's real -> real

File: shared/actions/chat.js [hunk 2]
```diff
@@ -536,40 +541,54 @@ function * _editMessage (action: EditMessage): SagaGenerator<any, any> {
...
-        identifyBehavior: TlfKeysTLFIdentifyBehavior.chatGui,
-        tlfName: users.join(','),
-        tlfVisibility: CommonTLFVisibility.private,
-        topicType: CommonTopicType.chat,
-      }})
-    const newConversationIDKey = result ? conversationIDToKey(result.conv.info.id) : null
-    if (!newConversationIDKey) {
-      console.warn('No convoid from newConvoRPC')
-      return null
+  const pendingTlfName = pendingConversationIDKeyToTlfName(oldConversationIDKey)
```
Comment (chrisnojima) [inline shared/actions/chat.js]: we handle pending or regular

File: shared/chat/conversation/index-hoc.js [hunk 3]
```diff
@@ -151,6 +151,6 @@ const editLastMessageHoc = compose(
   withProps(({setEditLastMessageCounter}) => ({onEditLastMessage: () => setEditLastMessageCounter(n => n + 1)}))
 )
 
-const hoc = compose(focusInputHoc, editLastMessageHoc, propsHoc, decoratedMesssagesHoc)
+const hoc = compose(focusInputHoc, editLastMessageHoc, decoratedMesssagesHoc, propsHoc)
```
Comment (chrisnojima) [inline shared/chat/conversation/index-hoc.js]: so we have to be really careful about the ordering of things in compose. What was happening was the items were being decorated (injecting the supersedes / superseded by stuff) but then it *wasn't* being passed to the children. This fixes that

File: shared/store/configure-store.js [hunk 4]
```diff
@@ -42,6 +42,9 @@ if (enableStoreLogging) {
     duration: true,
     logger: {
       error: () => {},
+      group: () => {},
```
Comment (chrisnojima) [inline shared/store/configure-store.js]: these were missing which was causing a lot of exceptions being thrown runtime

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready now

File: shared/actions/chat.js [hunk 5]
```diff
@@ -225,8 +230,8 @@ function openTlfInChat (tlf: string): OpenTlfInChat {
   return {type: 'chat:openTlfInChat', payload: tlf}
 }
 
-function startConversation (users: Array<string>): StartConversation {
-  return {type: 'chat:startConversation', payload: {users}}
+function startConversation (users: Array<string>, forceImmediate: ?boolean = false): StartConversation {
+  return {type: 'chat:startConversation', payload: {forceImmediate: !!forceImmediate, users}}
```
Comment (MarcoPolo) [inline shared/actions/chat.js]: You can just do:

`forceImmediate?: boolean = false)...` and then you don't need to do that!

File: shared/actions/chat.js (hunk 2 above)
Comment (MarcoPolo) [inline shared/actions/chat.js]: 🐑 🐑 This could be a separate function: `oldConvId  -> tlfName` with the null check below

**Comment by chrisnojima** [comment]
didn't hoist the tlfName out cause
1. it's only ever used by this function
2. we actually use  pendingTlfname below so we'd have to plumb that back also, sorta defeating the purpose of pulling it out

**Pull request merged**